src/hera/workflows/workflow.py

```diff
@@ -330,13 +332,44 @@ def to_yaml(self, *args, **kwargs) -> str:
         kwargs.setdefault("sort_keys", False)
         return _yaml.dump(self.to_dict(), *args, **kwargs)
 
-    def create(self) -> TWorkflow:
-        """Creates the Workflow on the Argo cluster."""
+    def create(self, async_: bool = True, poll_interval: int = 5) -> TWorkflow:
+        """Creates the Workflow on the Argo cluster.
+
+        Parameters
+        ----------
+        async_: bool = True
+            If async is true, then the workflow is created asynchronously and the function returns immediately.
+            If async is false, then the workflow is created and the function blocks until the workflow is done
+            executing.
+        poll_interval: int = 5
+            The interval in seconds to poll the workflow status if async is false. Ignored when async is true.
+        """
         assert self.workflows_service, "workflow service not initialized"
         assert self.namespace, "workflow namespace not defined"
-        return self.workflows_service.create_workflow(
+        if async_:
+            return self.workflows_service.create_workflow(
+                WorkflowCreateRequest(workflow=self.build()), namespace=self.namespace
+            )
+        return self.wait(poll_interval=poll_interval)
+
+    def wait(self, poll_interval: int = 5) -> TWorkflow:
```

Comment:
What do we think about a timeout parameter as well? We could then also allow a cleanup callback hook if it times outs and e.g. send a `WorkflowTerminateRequest` to kill the workflow

Reply:
Sounds like an amazing idea! What do you think about doing that in a follow up PR? We should also discuss how that will play with `active_deadline_seconds`: https://argoproj.github.io/argo-workflows/fields/#fields_3

Reply:
Sgtm! Won't be a breaking change so fine by me